frontend/app/routes/public/renew/$id/applicant-information.tsx

```diff
@@ -132,7 +132,7 @@ export async function action({ context: { session, serviceProvider }, params, re
   // TODO: handle when clientApplication is not found (null)
   invariant(clientApplication, 'Expected clientApplication to be found (not null).');
 
-  saveRenewState({ params, session, state: { applicantInformation: parsedDataResult.data } });
+  saveRenewState({ params, session, state: { applicantInformation: parsedDataResult.data, clientApplication: clientApplication } });
```

Comment:
Nitpick:  We can use the shorthand to reduce redundancy for `clientApplication` since the property name and value are the same

```suggestion
  saveRenewState({ params, session, state: { applicantInformation: parsedDataResult.data, clientApplication } });
```